### PR TITLE
Implement IngredientQuantity as a value and an optional unit

### LINKED ISSUE #52: Implement ingredient
Implement a data structure that represents an ingredient, consisting of the name and quantity of the ingredient.

- [x] #50 
- [x] #60

### PR DESCRIPTION
**Pull request by gohkh**
**tldr: For 2 ingredient quantities `a` and `b`, you can now execute `a.isCompatibleWith(b)` to check if they can be added/subtracted, then do `a.add(b)` or `a.subtract(b)`.**

Add Apache Commons Math 3.6.2 to gradle dependencies.

Add a new class `MixedFraction` to `seedu.address.commons.core.fraction`, which inherits from `Fraction` in Apache Commons Math.

Add parsing of ingredient quantity into a value and an optional unit.

Implement value as `BigDecimal` or `MixedFraction`. (`BigDecimal` is chosen to avoid floating point errors)

Store unit as a `String`.

Add support for addition and subtraction if the units are the same, and if the subtraction results in a negative value, return an `IngredientQuantity` with a value of 0.

Cap maximum denominator of fractions at 6, to avoid unusual fractions.

Add tests for `MixedFraction` and `IngredientQuantity`.

### CODE REVIEW

**Review by teo-jun-xiong — APPROVED** [review]
LGTM.

**Review by tzihiang — APPROVED** [review]
LGTM, thanks!

**Review by hans555 — APPROVED** [review]
